Added stuff (#310)
stackabletech/documentation: modules/ROOT/pages/release_notes.adoc

The Stackable platform consists of multiple operators that work together.
Periodically a platform release is made, including all components of the platform at a specific version.

== Release 22.11

> This is the third release of the Stackable Data Platform, it focuses on resource management.

Reviewer: ```suggestion
This is the third release of the Stackable Data Platform, which this time focuses on resource management.
```

* https://github.com/stackabletech/trino-operator/pull/310[Trino]
* https://github.com/stackabletech/airflow-operator/pull/174[Apache Airflow]

New Versions::
New product versions are supported.

* https://github.com/stackabletech/kafka-operator/pull/492[Apache Kafka 3.3.1]
* https://github.com/stackabletech/hdfs-operator/pull/250[Apache Hadoop HDFS 3.3.4]
* https://github.com/stackabletech/nifi-operator/pull/360[Apache NiFi 1.18.0]
* https://github.com/stackabletech/druid-operator/pull/317[Apache Druid 24.0.0]
* https://github.com/stackabletech/airflow-operator/pull/179[Apache Airflow 2.4.1]

Product features::
Additionally there are some individual product features that are noteworthy

* https://github.com/stackabletech/kafka-operator/pull/221[HBase: Phoenix support]
* https://github.com/stackabletech/hive-operator/pull/264[Hive: Support HDFS connection]
* https://github.com/stackabletech/nifi-operator/pull/323[NiFi: Support for in-place upgrades]
* https://github.com/stackabletech/nifi-operator/pull/371[NiFi: repository sizes are now adjusted based on declared PVC sizes]
* https://github.com/stackabletech/trino-operator/pull/306[Trino: Support for LDAP authentication]

> * Also the github repositories contain new and improved READMEs.

Reviewer: ```suggestion
* The github repositories contain new and improved READMEs.
```

=== Supported Kubernetes versions
This release supports the following Kubernetes versions:

* `1.25` (new)
* `1.24`
* `1.23`
* `1.22`

=== Upgrade from 22.09

==== Using stackablectl
You can list the available releases as follows

[source,console]
----
$ stackablectl release list
RELEASE            RELEASE DATE   DESCRIPTION
22.11              2022-11-08     Third release focusing on resource management
22.09              2022-09-09     Second release focusing on security and OpenShift support
22.06              2022-06-30     First official release of the Stackable Data Platform

----

To uninstall the `22.09` release run

[source,console]
----
$ stackablectl release uninstall 22.09
[INFO ] Uninstalling release 22.09
[INFO ] Uninstalling airflow operator
[INFO ] Uninstalling commons operator
# ...
----

Afterwards you will need to update the CustomResourceDefinitions (CRDs) installed by the Stackable Platform.

> The reason is, that helm will uninstall the operators but not the CRDs.

Reviewer: ```suggestion
The reason for this is that helm will uninstall the operators but not the CRDs.
```

[INFO ] Installing hdfs operator in version 0.6.0
[INFO ] Installing hive operator in version 0.8.0
[INFO ] Installing kafka operator in version 0.8.0
[INFO ] Installing nifi operator in version 0.8.0
[INFO ] Installing opa operator in version 0.11.0
[INFO ] Installing secret operator in version 0.6.0
[INFO ] Installing spark-k8s operator in version 0.6.0
[INFO ] Installing superset operator in version 0.7.0
[INFO ] Installing trino operator in version 0.7.0
[INFO ] Installing zookeeper operator in version 0.12.0
# ...
----

==== Using helm
Use `helm list` to list the currently installed operators.

You can use the following command to uninstall all of the operators that are part of the release 22.09:

[source,console]
----
$ helm uninstall airflow-operator commons-operator druid-operator hbase-operator hdfs-operator hive-operator kafka-operator nifi-operator opa-operator secret-operator spark-k8s-operator superset-operator trino-operator zookeeper-operator
release "airflow-operator" uninstalled
release "commons-operator" uninstalled
# ...
----

Afterwards you will need to update the CustomResourceDefinitions (CRDs) installed by the Stackable Platform.

> The reason is, that helm will uninstall the operators but not the CRDs.

Reviewer: ```suggestion
This is because helm will uninstall the operators but not the CRDs.
```

$ helm repo update stackable
$ helm install --wait airflow-operator stackable/airflow-operator --version 0.6.0
$ helm install --wait commons-operator stackable/commons-operator --version 0.4.0
$ helm install --wait druid-operator stackable/druid-operator --version 0.8.0
$ helm install --wait hbase-operator stackable/hbase-operator --version 0.5.0
$ helm install --wait hdfs-operator stackable/hdfs-operator --version 0.6.0
$ helm install --wait hive-operator stackable/hive-operator --version 0.8.0
$ helm install --wait kafka-operator stackable/kafka-operator --version 0.8.0
$ helm install --wait nifi-operator stackable/nifi-operator --version 0.8.0
$ helm install --wait opa-operator stackable/opa-operator --version 0.11.0
$ helm install --wait secret-operator stackable/secret-operator --version 0.6.0
$ helm install --wait spark-k8s-operator stackable/spark-k8s-operator --version 0.6.0
$ helm install --wait superset-operator stackable/superset-operator --version 0.7.0
$ helm install --wait trino-operator stackable/trino-operator --version 0.7.0
$ helm install --wait zookeeper-operator stackable/zookeeper-operator --version 0.12.0
----

==== Breaking changes

> You will need to adapt your existing CRDs to the following breaking changes:

Reviewer: ```suggestion
You will need to adapt your existing CRDs due to the following breaking changes:
```

$ helm install --wait commons-operator stackable/commons-operator --version 0.4.0
$ helm install --wait druid-operator stackable/druid-operator --version 0.8.0
$ helm install --wait hbase-operator stackable/hbase-operator --version 0.5.0
$ helm install --wait hdfs-operator stackable/hdfs-operator --version 0.6.0
$ helm install --wait hive-operator stackable/hive-operator --version 0.8.0
$ helm install --wait kafka-operator stackable/kafka-operator --version 0.8.0
$ helm install --wait nifi-operator stackable/nifi-operator --version 0.8.0
$ helm install --wait opa-operator stackable/opa-operator --version 0.11.0
$ helm install --wait secret-operator stackable/secret-operator --version 0.6.0
$ helm install --wait spark-k8s-operator stackable/spark-k8s-operator --version 0.6.0
$ helm install --wait superset-operator stackable/superset-operator --version 0.7.0
$ helm install --wait trino-operator stackable/trino-operator --version 0.7.0
$ helm install --wait zookeeper-operator stackable/zookeeper-operator --version 0.12.0
----

==== Breaking changes
You will need to adapt your existing CRDs to the following breaking changes:

> ==== Stackable operator for Apache Spark

Reviewer 2: ```suggestion
==== Stackable Operator for Apache Spark
```
